fix: update permissionless paymaster tutorial & l1-l2 (#47)
zkSync-Community-Hub/community-code: content/tutorials/permissionless-paymaster/10.index.md

![manager-signer-relation-diagram](/images/permissionless-paymaster/manager-signer.jpg)

## Integration
The diagram below provides the flow of the integration:

1. Dapp decides on custom logic for each user. Let's assume that Dapp decides to sponsor gas for every approved transaction.

2. Dapp calls the backend server or Zyfi API with relevant data to get the signer's signature.

>     For the utmost security, it is recommended that the signer's signature be sent to Dapp's secure backend server.

Reviewer: ```suggestion
    For security best-practices, it is recommended to create the signer's signature on a secure backend server.
```

3. The signer's key signs this paymaster data and returns the signature and signer address to the Dapp's frontend.

4. Paymaster address and required data with signature are added to the transaction blob in the frontend.

> 5. The user receives the transaction signature request pop-up on their wallet. **The User only signs the transaction**, and it is sent on-chain.

Reviewer 2: Change `**The User` to `**The user` for consistency

    const innerInput = ethers.utils.arrayify(
      abiCoder.encode(["uint256", "uint256", "address", "bytes"],
        [
          expirationTime, // As used in the above signature
          maxNonce, // As used in the above signature
          signerAddress, // The signer address
          signature,
        ]
      ) // Signature created in the above snippet. get from API server
    );
    // getPaymasterParams function is available in zksync-ethers
    const paymasterParams = utils.getPaymasterParams(
        paymasterAddress, // Paymaster address
        {
            type: "General",
            innerInput: innerInput,
        }
    );
    // Returns paymaster params, gas fee, gas limit
    return [paymasterParams, maxFeePerGas, gasLimit];
 };
```

##### **5.2** Estimate gas and call the above function

In the `WriteContract` component, right before the `const tx = await contract.approve(spender, amount);` line, we will:

1. Estimate gas
2. Call the `preparePaymasterParam` function we created above

with the snippet below:

```javascript [src/components/WriteContract.tsx]
   // Estimate gas
    const estimateGas = await contract.estimateGas.approve(spender,amount); 

>     const [account, paymasterParams, maxFeePerGas, gasLimit] = await preparePaymasterParam(account, estimateGas);

Reviewer 3: This would lead to error

   // Estimate gas
    const estimateGas = await contract.estimateGas.approve(spender,amount); 

>     const [account, paymasterParams, maxFeePerGas, gasLimit] = await preparePaymasterParam(account, estimateGas);

Reviewer 3: This would lead to error.
